test/tests/functional/pbs_node_rampdown_cset.py

```diff
@@ -0,0 +1,175 @@
+        # three from cpuset system (natural + 2 NUMA vnodes)
+        nodeinfo = self.server.status(NODE)
+        if len(nodeinfo) < 4:
+            self.skipTest("Not enough vnodes to run the test.")
+        # skip if second mom has less than two NUMA vnodes
+        if nodeinfo[3]['id'] != self.n3:
+            self.skipTest("Second mom has less than 2 vnodes")
+        # skip if none of the vnodes are in free state
+        for node in nodeinfo:
+            if node['state'] != 'free':
+                self.skipTest("Not all the vnodes are in free state")
+
+        self.pbs_release_nodes_cmd = os.path.join(
+            self.server.pbs_conf['PBS_EXEC'], 'bin', 'pbs_release_nodes')
+
+        # number of resource ncpus to request initially
+        ncpus = self.server.status(NODE, 'resources_available.ncpus',
+                                   id=self.n3)[0]
+        ncpus = int(ncpus['resources_available.ncpus'])
+        self.ncpus2 = ncpus / 2
+
+        # expected values upon successful job submission
+        self.job1_schedselect = \
+            "1:ncpus=1:mem=2gb+" + \
+            "1:ncpus=%d:mem=2gb:vnode=%s+" % (self.ncpus2, self.n2) + \
+            "1:ncpus=%d:mem=2gb:vnode=%s" % (self.ncpus2, self.n3)
+        self.job1_exec_host = "%s/0+%s/0*%d+%s/1*%d" % (
+            self.n0, self.n1, self.ncpus2, self.n1, self.ncpus2)
+        self.job1_exec_vnode = \
+            "(%s:ncpus=1:mem=2097152kb)+" % (self.n0,) + \
+            "(%s:ncpus=%d:mem=2097152kb)+" % (self.n2, self.ncpus2) + \
+            "(%s:ncpus=%d:mem=2097152kb)" % (self.n3, self.ncpus2)
+
+        # expected values after release of vnode
+        self.job1_schedsel1 = \
+            "1:ncpus=1:mem=2097152kb+" + \
+            "1:ncpus=%d:mem=2097152kb:vnode=%s" % (self.ncpus2, self.n2)
+        self.job1_exec_host1 = \
+            "%s/0+%s/0*%d" % (self.n0, self.n1, self.ncpus2)
+        self.job1_exec_vnode1 = \
+            "(%s:ncpus=1:mem=2097152kb)+" % (self.n0,) + \
+            "(%s:ncpus=%d:mem=2097152kb)" % (self.n2, self.ncpus2)
+
+        # cgroup cpuset path on second node
+        cmd = ['grep cgroup', '/proc/mounts', '|', 'grep cpuset', '|',
+               'grep -v', '/dev/cpuset']
+        ret = self.server.du.run_cmd(self.n1, cmd, runas=TEST_USER)
+        self.cset_path = ret['out'][0].split()[1]
+
+    def test_release_nodes_on_cpuset_sis(self):
+        """
+        Submit job that will use cpus on two NUMA vnodes on second mom,
+        goes in R state. Use pbs_release_node to successfully release one
+        of the NUMA vnodes and its resources used in the job.
+        """
+        # Submit a job that uses second mom's two NUMA nodes, in R state
+        a = {'Resource_List.select': 'ncpus=1:mem=2gb+' +
+             'ncpus=%d:mem=2gb:vnode=%s+' % (self.ncpus2, self.n2) +
+             'ncpus=%d:mem=2gb:vnode=%s' % (self.ncpus2, self.n3),
+             'Resource_List.place': 'vscatter'}
```

Comment:
Now that you request the NUMA node vnodes specifically, is the "vscatter" still necessary because you want the first chunk to not be put on the 2 NUMA nodes?

Reply:
Without the vscatter there is a chance that the first chunk uses one of the NUMA vnodes having sharing=default_shared. I wanted the job's exec_vnode to use the first mom for the first chunk.